airbyte/_executors/python.py

```diff
@@ -42,6 +44,11 @@ def __init__(
             pip_url: (Optional.) The pip URL of the connector to install.
             install_root: (Optional.) The root directory where the virtual environment will be
                 created. If not provided, the current working directory will be used.
+            use_python: (Optional.) Python interpreter specification:
+                - True: Use current Python interpreter
+                - False: Use Docker instead (handled by factory)
+                - Path: Use interpreter at this path
+                - str: Use uv-managed Python version
```

Comment:
It's not obvious in this convention whether 'python3.12' is a an already installed python version or the request for a uv-managed python version. Modify this so we treat the string as a uv-managed version if it matches a semver version (excluding the 'python' prefix) and as an expected-to-exist python interpreter if any other string is provided (e.g. 'python3.10' or '/path/to/python3.10')